src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -51,6 +52,32 @@ Logger frontend_logger;
 std::set<ProcessInfo> processData;
 std::mutex aggregateWeightMutex;
 std::mutex triangleTreeMutex;
+// Thread function
+void listen_to_kafka_topic(KafkaConnector &kstream,Partitioner &graphPartitioner,vector<DataPublisher*> &workerClients)
+{
+    while (true) {
+        cppkafka::Message msg = kstream.consumer.poll();
+        if (!msg || msg.get_error()) {
+            continue;
+        }
+        string data(msg.get_payload());
+        if (data == "-1") {  // Marks the end of stream
+            frontend_logger.log("Received the end of stream", "info");
```

Comment:
Can we provide more informative error message here? `"Received the end of stream"` does not include on which stream we have found the end of stream.

Reply:
Ok sir